tests/zfs-tests/cmd/randfree_file/randfree_file.c

```diff
@@ -98,11 +98,13 @@ main(int argc, char *argv[])
 
 	free(buf);
 
+#if defined(FALLOC_FL_PUNCH_HOLE) && defined(FALLOC_FL_KEEP_SIZE)
 	if (fallocate(fd, FALLOC_FL_PUNCH_HOLE | FALLOC_FL_KEEP_SIZE,
 	    start_off, off_len) < 0) {
 		perror("fallocate");
 		return (1);
 	}
+#endif /* defined(FALLOC_FL_PUNCH_HOLE) && defined(FALLOC_FL_KEEP_SIZE) */
```

Comment:
I have submitted the same code twice, but the CI has different failures log , why? The CI environment is unstable? Please help me, thank you @behlendorf